crates/sui-types/src/execution.rs

```diff
@@ -167,6 +167,23 @@ impl ExecutionResultsV2 {
                 }
             }
 
+            // Record start version for ConsensusV2 objects.
+            if let Owner::ConsensusV2 { start_version, .. } = &mut obj.owner {
+                debug_assert!(!self.deleted_object_ids.contains(id));
+
+                if let Some(Owner::ConsensusV2 {
+                    start_version: previous_start_version,
+                    ..
+                }) = input_objects.get(id).map(|obj| &obj.owner)
+                {
+                    *start_version = *previous_start_version;
```

Comment:
Is there any reason that `start_version` may not be equal to `previous_start_version` here?

Reply:
Not that I can think of, but I copied that logic from the shared object case above to be safe

Is there a reason that it can differ for the shared objects or could all of this be simplified?

Reply:
I believe the answer is "reshares" but maybe @tzakian would remember

Reply:
Right that makes sense. In this case "reshare" may not even be dumb/extraneous, because you could be transferring it to a new owner/authenticator but still keeping the same ownership type

Reply:
@tnowacki Could you remind me what "reshare" means here?
In anyway it would be good to document the context as comment there. I am sure I will forget about this in a few months.

Reply:
Consider this move function
```
public fun foo(o: MySharedObj) {
    sui::transfer::share_object(o)
}
```
A "reshare" is when `foo` is called with an input object that is a shared object (and it couldn't be any other kind of input object because we do not support upgrades currently)

This case became possible once we allowed shared objects to be taken by-value to enable shared object deletion.

Reply:
@lxfind added a comment per your suggestion